File: 7-7/backend/schemas/face.py
from typing import List, Optional
from pydantic import BaseModel, Field


class FaceRegisterRequest(BaseModel):
    image: str = Field(..., description="Base64 encoded image")


class FaceVerifyRequest(BaseModel):
    image: str = Field(..., description="Base64 encoded image")


class FaceDetectRequest(BaseModel):
    image: str = Field(..., description="Base64 encoded image")


class FaceDetectResponse(BaseModel):
    detected: bool
    x: Optional[int] = None
    y: Optional[int] = None
    w: Optional[int] = None
    h: Optional[int] = None


class FaceEmbeddingResponse(BaseModel):
    id: int
    userId: int
    imagePath: Optional[str]
    createdAt: str

    class Config:
        from_attributes = True


class FaceVerifyResponse(BaseModel):
    verified: bool
    similarity: float
    message: str


class FaceVerifyPreviewRequest(BaseModel):
    image: str = Field(..., description="Base64 encoded image")


class FaceVerifyPreviewResponse(BaseModel):
    detected: bool
    similarity: float
    verified: bool

File: 15-1/frontend/src/api/face.ts
import apiClient from './client'

export interface FaceRegisterRequest {
  image: string
}

export interface FaceEmbeddingResponse {
  id: number
  userId: number
  imagePath: string | null
  createdAt: string
}

export interface FaceDetectRequest {
  image: string
}

export interface FaceDetectResponse {
  detected: boolean
  x?: number
  y?: number
  w?: number
  h?: number
}

export const registerFaceBase64 = async (data: FaceRegisterRequest): Promise<FaceEmbeddingResponse> => {
  const response = await apiClient.post<FaceEmbeddingResponse>('/face/register-base64', data)
  return response.data
}

export const getFaceEmbeddings = async (): Promise<FaceEmbeddingResponse[]> => {
  const response = await apiClient.get<FaceEmbeddingResponse[]>('/face/embeddings')
  return response.data
}

export const detectFace = async (data: FaceDetectRequest): Promise<FaceDetectResponse> => {
  const response = await apiClient.post<FaceDetectResponse>('/face/detect', data)
  return response.data
}

export const detectFacePublic = async (data: FaceDetectRequest): Promise<FaceDetectResponse> => {
  const response = await apiClient.post<FaceDetectResponse>('/face/detect/public', data)
  return response.data
}

export interface FaceVerifyPreviewRequest {
  image: string
}

export interface FaceVerifyPreviewResponse {
  detected: boolean
  similarity: number
  verified: boolean
}

export const verifyFacePreview = async (
  data: FaceVerifyPreviewRequest,
): Promise<FaceVerifyPreviewResponse> => {
  const response = await apiClient.post<FaceVerifyPreviewResponse>('/face/verify-preview', data)
  return response.data
}
